tests/operator/operator.go

```diff
@@ -3404,3 +3339,37 @@ func getDaemonsetImage(name string) string {
 
 	return matches[0][2]
 }
+
+func getComponentConfigPatchOption(toChange *v1.ComponentConfig, orig *v1.ComponentConfig, field string) patch.PatchOption {
+	path := "/spec/" + field
+	var opt patch.PatchOption
+	if toChange == nil {
+		opt = patch.WithRemove(path)
+	} else {
+		if orig != nil {
+			opt = patch.WithReplace(path, toChange)
+		} else {
+			opt = patch.WithAdd(path, toChange)
```

Comment:
The "add" operation will also update a value if it already exists. The only reason to use "replace" here is if we want the patch to fail if the original value does not exist, but since you check it here in-code I'm not sure that's valuable.

```suggestion
			opt = patch.WithAdd(path, toChange)
```

Reply:
I think it's clearer this way.